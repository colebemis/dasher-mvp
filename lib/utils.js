import Router from 'next/router'

export function redirect(target, res) {
  if (res) {
    // Server-side
    res.writeHead(302, { Location: target })
    res.end()
  } else {
    // Client-side
    Router.replace(target)
  }
}
