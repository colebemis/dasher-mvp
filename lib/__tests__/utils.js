import Router from 'next/router'
import { redirect } from '../utils'

describe('redirect', () => {
  it('redirects server-side', () => {
    const res = {
      writeHead: jest.fn(),
      end: jest.fn(),
    }

    redirect('/foo', res)

    expect(res.writeHead.mock.calls).toMatchSnapshot()
  })

  it('redirects client-side', () => {
    redirect('/foo')

    expect(Router.replace).toHaveBeenCalledWith('/foo')
  })
})
