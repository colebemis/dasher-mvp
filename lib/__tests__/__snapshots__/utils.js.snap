// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redirect redirects server-side 1`] = `
Array [
  Array [
    302,
    Object {
      "Location": "/foo",
    },
  ],
]
`;
